app/data/bash-completion/scrcpy

```diff
@@ -93,6 +93,11 @@ _scrcpy() {
             COMPREPLY=($(compgen -W 'verbose debug info warn error' -- "$cur"))
             return
             ;;
+        -s|--serial)
+            # Use 'adb devices' to list serial numbers.
+            COMPREPLY=($(compgen -W "$(adb devices | awk '/(device|recovery|sideload)$/{print $1}')" -- ${cur} ) )
```

Comment:
I think that scrcpy may not work in recovery or sideload state, so maybe `adb devices |awk '/device$/{print $1}'` is sufficient.